openfortivpn-webview-qt/mainwindow.cpp

```diff
@@ -17,7 +17,7 @@ MainWindow::MainWindow(const bool keepOpen,
                        const QRegularExpression& urlToWaitForRegex,
                        QWidget *parent) :
     QMainWindow(parent),
-    webEngine(new QWebEngineView(parent)),
+    webEngine(new QWebEngineView(new QWebEngineProfile("vpn", parent), parent)),
```

Comment:
I'm looking at the [documentation](https://doc.qt.io/qt-6/qwebengineview.html#QWebEngineView-1) where it says:

> Note: The profile object ownership is not taken and it should outlive the view.

With your change, it does outlive the view, but it is also never deleted.

You need something like this [here](https://github.com/gm-vm/openfortivpn-webview/blob/48aab35f79c0feca3bfc9f79cf266dbf796b1040/openfortivpn-webview-qt/mainwindow.h#L26):
```c++
private:
    QWebEngineProfile *webEngineProfile;
    QWebEngineView *webEngine;
```

Then you can have this here:
```c++
MainWindow::MainWindow(const bool keepOpen,
                       const QRegularExpression& urlToWaitForRegex,
                       QWidget *parent) :
    QMainWindow(parent),
    webEngineProfile("vpn", parent),
    webEngine(webEngineProfile),
```

Last, but not least:
```c++
MainWindow::~MainWindow()
{
    delete webEngine;
    delete webEngineProfile;
}
```

This last step is also important because:
> A disk-based QWebEngineProfile should be destroyed on or before application exit, otherwise the cache and persistent data may not be fully flushed to disk.

Reply:
Updated. I had to use constructors though instead of just parameters when initializing profile/view, so it's different from your direct suggestion.

Reply:
Yes, sorry, I meant to write what you did.